Library/Formula/lcm.rb

```diff
@@ -0,0 +1,48 @@
+class Lcm < Formula
+  desc "libraries and tools for message passing and data marshalling"
+  homepage "https://lcm-proj.github.io/"
+  url "https://github.com/lcm-proj/lcm/releases/download/v1.3.0/lcm-1.3.0.zip"
+  sha256 "23bce19a4a6dc5df444d811f57796e14b4b0d1886cb3c219f90b72f328f2184e"
+
+  head do
+    # This url can be changed to "https://github.com/lcm-proj/lcm.git"
+    # after PR #48 on lcm is accepted.
+    url "https://github.com/bluesquall/lcm.git", :branch => "simplify-java-Makefile.am"
+
+    depends_on "xz" => :build
+    depends_on "libtool" => :build
+    depends_on "automake" => :build
+    depends_on "autoconf" => :build
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on :java => :recommended
```

Comment:
Or is that just the expected behavior of `:optional` `:java` dependencies, and the build and run time aspects of the dependency are understood to be independent? Still weird that it means we end up with a bottle that does have Java support but you have to know not to explicitly ask for `--with-java`. If this is the case, then `:optional` makes sense because it means that most everybody gets Java support anyway.

Do you know what the failure mode of LCM is if a Java-enabled build is installed on a machine without Java? Is that just the same as doing a non-Java build, and you can just install Java without reinstalling LCM to enable the Java-dependent bits?